src/test/java/integ/com/mathworks/ci/TestResultsViewActionTest.java

```diff
@@ -0,0 +1,427 @@
+        int actualCount = ac.getTotalCount();
+        Assert.assertEquals("Incorrect total tests count",4,actualCount);
+    }
+
+    /**
+     *  Verify if passed tests count is correct
+     *
+     */
+
+    @Test
+    public void verifyPassedTestsCount() throws ExecutionException, InterruptedException, URISyntaxException, IOException, ParseException {
+        FreeStyleBuild build = getFreestyleBuild();
+        final FilePath workspace = new FilePath(new File("C:", "workspace"));
+        final String actionID = "abc123";
+        final String targetFile = MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + actionID + ".json";
+        FilePath artifactRoot = new FilePath(build.getRootDir());
+        copyFileInWorkspace("testArtifacts/t1/windows/" + MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + ".json",targetFile,artifactRoot);
+        TestResultsViewAction ac = new TestResultsViewAction(build, workspace, actionID);
+        int actualCount = ac.getPassedCount();
+        Assert.assertEquals("Incorrect passed tests count",2,actualCount);
+    }
+
+    /**
+     *  Verify if failed tests count is correct
+     *
+     */
+
+    @Test
+    public void verifyFailedTestsCount() throws ExecutionException, InterruptedException, URISyntaxException, IOException, ParseException {
+        FreeStyleBuild build = getFreestyleBuild();
+        final FilePath workspace = new FilePath(new File("C:", "workspace"));
+        final String actionID = "abc123";
+        final String targetFile = MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + actionID + ".json";
+        FilePath artifactRoot = new FilePath(build.getRootDir());
+        copyFileInWorkspace("testArtifacts/t1/windows/" + MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + ".json",targetFile,artifactRoot);
+        TestResultsViewAction ac = new TestResultsViewAction(build, workspace, actionID);
+        int actualCount = ac.getFailedCount();
+        Assert.assertEquals("Incorrect failed tests count",1,actualCount);
+    }
+
+    /**
+     *  Verify if incomplete tests count is correct
+     *
+     */
+
+    @Test
+    public void verifyIncompleteTestsCount() throws ExecutionException, InterruptedException, URISyntaxException, IOException, ParseException {
+        FreeStyleBuild build = getFreestyleBuild();
+        final FilePath workspace = new FilePath(new File("C:", "workspace"));
+        final String actionID = "abc123";
+        final String targetFile = MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + actionID + ".json";
+        FilePath artifactRoot = new FilePath(build.getRootDir());
+        copyFileInWorkspace("testArtifacts/t1/windows/" + MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + ".json",targetFile,artifactRoot);
+        TestResultsViewAction ac = new TestResultsViewAction(build, workspace, actionID);
+        int actualCount = ac.getIncompleteCount();
+        Assert.assertEquals("Incorrect incomplete tests count",0,actualCount);

```

Comment:
We should maybe test with a non-default value instead of 0?

Reply:
Thanks for catching that, updated the test!